lib/systems/examples.nix

```diff
@@ -93,6 +93,26 @@ rec {
     config = "mipsel-unknown-linux-gnu";
   } // platforms.fuloong2f_n32;
 
+  # MIPS ABI table transcribed from here: https://wiki.debian.org/Multiarch/Tuples
+
+  # can execute on 32bit chip
+  mips-linux-gnu                = { config = "mips-linux-gnu";                } // platforms.gcc_mips32r2_o32;
+  mipsel-linux-gnu              = { config = "mipsel-linux-gnu";              } // platforms.gcc_mips32r2_o32;
+  mipsisa32r6-linux-gnu         = { config = "mipsisa32r6-linux-gnu";         } // platforms.gcc_mips32r6_o32;
```

Comment:
This also does not evaluate.

Reply:
See above.

Reply:
As an aside, it would be really nice if nixpkgs had a template "here's what you need to fill in to add support for a new platform", all in one file.  Right now it's scattered all over the place in `lib/systems/` (and elsewhere too).

Then it would be obvious whether things like this are checked in because we were transcribing some table for the benefit of future platforms-to-be-added, or whether full platform support is intended.

A nixpkgs "platform" is really just a gnu 5-tuple (cpu, vendor, kernel, libc, abi).  Everything else (kernel flags, compiler optimization flags, etc) are site-specific customizations which don't affect interoperatbility between different derivations.

I would advise eliminating any distinction between "platform" and "gnu 5-tuple".

Then, for a given product that can be bought (say, a Raspberry Pi), all you need to know is which *kernel* to use.  Kernel configuration is much more complex than just a 5-tuple, and will vary among different devices which belong to the same 5-tuple.  Any given kernel will support some set of one or more 5-tuples (for example, amd64 machines can run i386-linux-gnu userspaces as well as x86_64-linux-musl userspaces).  Once you've picked your hardware and picked your kernel, you pick your platform, then you're done.